src/meta/processors/job/DataBalanceJobExecutor.cpp

```diff
@@ -0,0 +1,197 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include "meta/processors/job/DataBalanceJobExecutor.h"
+
+#include <folly/executors/CPUThreadPoolExecutor.h>
+
+#include "common/utils/MetaKeyUtils.h"
+#include "kvstore/NebulaStore.h"
+#include "meta/processors/job/JobUtils.h"
+
+namespace nebula {
+namespace meta {
+
+folly::Future<Status> DataBalanceJobExecutor::executeInternal() {
+  if (plan_ == nullptr) {
+    Status status = buildBalancePlan();
+    if (status != Status::OK()) {
+      return status;
+    }
+  }
+  plan_->setFinishCallBack([this](meta::cpp2::JobStatus status) {
+    if (LastUpdateTimeMan::update(kvstore_, time::WallClock::fastNowInMilliSec()) !=
+        nebula::cpp2::ErrorCode::SUCCEEDED) {
+      LOG(ERROR) << "Balance plan " << plan_->id() << " update meta failed";
+    }
+    executorOnFinished_(status);
+  });
+  plan_->invoke();
+  return Status::OK();
+}
+
+Status DataBalanceJobExecutor::buildBalancePlan() {
+  std::map<std::string, std::vector<Host*>> lostZoneHost;
+  std::map<std::string, std::vector<Host*>> activeSortedHost;
+  for (auto& zoneMapEntry : spaceInfo_.zones_) {
+    for (auto& hostMapEntry : zoneMapEntry.second.hosts_) {
+      activeSortedHost[zoneMapEntry.first].push_back(&hostMapEntry.second);
+    }
+  }
+  for (HostAddr hosts : lostHosts_) {
+    if (!spaceInfo_.hasHost(hosts)) {
+      return Status::Error(
+          "Host %s does not belong to space %d", hosts.toString().c_str(), spaceInfo_.spaceId_);
+    }
+    for (auto& zoneMapEntry : spaceInfo_.zones_) {
+      auto it = zoneMapEntry.second.hosts_.find(hosts);
+      if (it != zoneMapEntry.second.hosts_.end()) {
+        lostZoneHost[zoneMapEntry.first].push_back(&it->second);
+        std::vector<Host*>& hvec = activeSortedHost[zoneMapEntry.first];
+        hvec.erase(std::find(hvec.begin(), hvec.end(), &it->second));
+        break;
+      }
+    }
+  }
+  for (auto& hostMapEntry : activeSortedHost) {
+    std::vector<Host*>& hvec = hostMapEntry.second;
+    std::sort(hvec.begin(), hvec.end(), [](Host*& l, Host*& r) -> bool {
+      return l->parts_.size() < r->parts_.size();
+    });
+  }
+  plan_.reset(new BalancePlan(jobDescription_, kvstore_, adminClient_));
+  // move parts of lost hosts to active hosts in the same zone
+  for (auto& zoneHostEntry : lostZoneHost) {
```

Comment:
Please use some meaningful names, it's really suffering to read this kind of code.

for example:
```c++
for (auto& [zname, lostHosts] : lostZoneHost) {
  std::vector<Host*>& ativeHosts = activeSortedHost[zname];
  for (Host* lostHost : lostHosts) {
```

Reply:
done